### PR TITLE
Allow `CrosshairTool` and `HoverTool` to trigger from axes

### LINKED ISSUE #14122: Allow `CrosshairTool` to work on axes
Currently `CrosshairTool` only works within the frame. It may be useful in some scenarios to allow it to work on axes as well. In that case, it would only show one of the spans, depending on the dimension and of course if configured.

### PR DESCRIPTION
**Pull request by mattpap**
[Screencast from 2024-10-26 01-04-07.webm](https://github.com/user-attachments/assets/cb66bcb5-c66e-4ce5-9bab-eb071e58cc9c)

fixes #14122

### CODE REVIEW

**Comment by mattpap** [comment]
I added similar support to `HoverTool`:

[Screencast from 2024-10-26 10-40-53.webm](https://github.com/user-attachments/assets/50622fae-c6a0-456c-89c1-63d35aa6602a)

File: docs/bokeh/source/docs/releases/3.7.0.rst [hunk 1]
```diff
@@ -7,3 +7,4 @@ Bokeh version ``3.7.0`` (January 2025) is a minor milestone of Bokeh project.
 
 * Improved ``PanTool``'s cursor and state management handling (:bokeh-pull:`14106`)
 * Improved handling of ``touch-action`` based on active tools and their supported events (:bokeh-pull:`14109`)
+* Allowed ``CrosshairTool`` and ``HoverTool`` to work on axes (:bokeh-pull:`14124`)
```
Comment (bryevdv) [inline docs/bokeh/source/docs/releases/3.7.0.rst]: "work on" sort of implies to me that a crosshair or tooltip will pop up on the axis itself. Maybe "trigger from axes" or similar?

File: bokehjs/src/lib/models/tools/inspectors/crosshair_tool.ts [hunk 2]
```diff
@@ -76,12 +76,26 @@ export class CrosshairToolView extends InspectToolView {
     }
 
     const {sx, sy} = ev
-
-    if (!this.plot_view.frame.bbox.contains(sx, sy)) {
-      this._update_spans(NaN, NaN)
-    } else {
+    if (this.plot_view.frame.bbox.contains(sx, sy)) {
       this._update_spans(sx, sy)
+      return
```
Comment (bryevdv) [inline bokehjs/src/lib/models/tools/inspectors/crosshair_tool.ts]: Usually fine with early returns but I like them less when there's not an actual value. I guess the logic could be pulled into a function that returns `[sx, sy]` then just have one call to `this._update_spans(sx, sy)` after.

**Review by bryevdv — APPROVED** [review]
Just minor comments

**Comment by mattpap** [comment]
Test failures are unrelated and reported in #14128.

**Pull request merged**